OpenIPSL/Electrical/Sources/SourceBehindImpedance/VoltageSources/VSourceIO.mo

```diff
@@ -0,0 +1,117 @@
+          fillColor={217,67,180},
+          fillPattern=FillPattern.Solid,
+          radius=15), Text(
+          extent={{-80,-18},{80,-98}},
+          textColor={217,67,180},
+          textString="Text Layer assigns p2R.y_re and p2R.y_im to uvre and uvim, 
+which are the real and imaginary input
+ voltage deviations from the initial value")}), Icon(graphics={
+        Ellipse(extent={{-100,100},{100,-100}},
+                                           lineColor={215,215,215},
+          lineThickness=1,
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid),
+        Line(
+          points={{-100,60},{-60,60}},
+          color={28,108,200},
+          thickness=1),
+        Polygon(
+          points={{3,8},{7,-2},{-1,-2},{3,8}},
+          lineColor={28,108,200},
+          lineThickness=1,
+          fillColor={28,108,200},
+          fillPattern=FillPattern.Solid,
+          origin={-48,3},
+          rotation=270),
+        Line(
+          points={{-60,60},{-60,0}},
+          color={28,108,200},
+          thickness=1),
+        Line(
+          points={{-100,-60},{-60,-60}},
+          color={28,108,200},
+          thickness=1),
+        Line(
+          points={{-60,-60},{-60,0}},
+          color={28,108,200},
+          thickness=1),
+        Line(
+          points={{-60,0},{-40,0}},
+          color={28,108,200},
+          thickness=1),
+        Ellipse(extent={{-40,40},{40,-40}},lineColor={28,108,200},
+          lineThickness=1,
+          fillColor={255,255,255},
+          fillPattern=FillPattern.Solid),
+        Ellipse(
+          extent={{-20,-10},{0,10}},
+          lineColor={28,108,200},
+          lineThickness=1,
+          fillColor={170,213,255},
+          fillPattern=FillPattern.Solid,
+          startAngle=0,
+          endAngle=180,
+          closure=EllipseClosure.None),
+        Ellipse(
+          extent={{0,10},{20,-10}},
+          lineColor={28,108,200},
+          lineThickness=1,
+          fillColor={170,213,255},
+          fillPattern=FillPattern.None,
+          startAngle=0,
+          endAngle=180,
+          closure=EllipseClosure.None),
+        Line(
+          points={{40,0},{100,0}},
+          color={0,0,255},
+          thickness=1),
+        Text(
+          extent={{-100,140},{100,100}},
+          textColor={0,0,0},
+          textString="%name")}),
+    Documentation(info="<html>
+<p>
+This model provides a voltage source with an internal voltage source and internal impedance whose magnitude and angle can be varied via inputs starting from their initial values. 
+
+The initial values of the internal voltage source magnitude and angle are calculated based on the impedance and power flow data. 
+
+It is based on [Du2021], with the difference that the internal impedance includes a resistance and not only a reactance. 
+
+The purpose of this model is to support the development of Grid-Forming Inverter models as described in [Du2021].
+</p>
```

Comment:
Are the blank lines just to make it easier to edit? Since for reading you would need to put them into p env each.

Reply:
Yes, it was just easier to type, I've removed them.

Reply:
That was not the intention of my question. It is fine to keep blank lines (and usage of short <80 char lines) for better readability. It just looked liked you intended to have the line breaks also in the documentation (which you don't as I understand now). So best to revert this again.